### DISCUSSION TITLE
Kibana Error After Update

### DISCUSSION THREAD

**Discussion by xaqpuyrthnbg** [Security-Onion-Solutions/securityonion#3593] [Unsupported Versions] [answered]
Good morning, 

I updated to the current version of SO and lost Kibana again.  The last time I was able to follow the troubleshooting guide in the docs and the issue was resolved.  However, this time I seem to be getting a licensing error in /opt/so/log/kibana/kibana.log:
`{"type":"log","@timestamp":"2021-03-23T00:10:00+00:00","tags":["warning","plugins","licensing"],"pid":293,"message":"License information could not be obtained from Elasticsearch due to Error: Cluster client cannot be used after it has been closed. error"}`

Also restarted Kibana and rebooted.  This is the error after Kibana restart:

```
Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (300/300)
                  Server is not ready
                  Server still not ready after 300 attempts; giving up.
```

I did not see any references to this error in the discussion quite yet, but I do remember agreeing on the ES screen during the upgrade.  

Thanks for the help.

**Comment by TOoSmOotH** [comment]
What does the elastic log say?

**Comment by xaqpuyrthnbg** [comment]
tail of elastic log:
`[2021-03-23T12:05:25,771][DEPRECATION][org.elasticsearch.deprecation.cluster.metadata.IndexNameExpressionResolver] this request accesses system indices: [.kibana_1, .kibana_2, .kibana_task_manager_1, .tasks], but in a future major version, direct access to system indices will be prevented by default
[2021-03-23T12:05:28,644][DEPRECATION][org.elasticsearch.deprecation.cluster.metadata.IndexNameExpressionResolver] this request accesses system indices: [.kibana_1, .kibana_2, .kibana_task_manager_1, .tasks], but in a future major version, direct access to system indices will be prevented by default
[2021-03-23T12:05:28,644][DEPRECATION][org.elasticsearch.deprecation.cluster.metadata.IndexNameExpressionResolver] this request accesses system indices: [.kibana_1, .kibana_2, .kibana_task_manager_1, .tasks], but in a future major version, direct access to system indices will be prevented by default
[root@securityonionv3 elasticsearch]# `

**Reply by TOoSmOotH** [reply]
Did you customize your kibana config and move it to local?

**Comment by xaqpuyrthnbg** [comment]
The only changes to Kibana specifically were deleting the indices after the last upgrade to get it running again.

**Reply by TOoSmOotH** [reply]
Do you have anything in  /opt/so/saltstack/local/salt/kibana/

**Reply by xaqpuyrthnbg** [reply]
(4) folders but no files inside any of them; /bin /custom /etc /files.

**Reply by TOoSmOotH** [reply]
ok try sudo so-saltstack-update then run sudo so-kibana-restart

**Reply by xaqpuyrthnbg** [reply]
I will give it some additional time but this is the result:

` Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (300/300)
                  Server is not ready
                  Server still not ready after 300 attempts; giving up.
`

**Reply by TOoSmOotH** [reply]
I don't understand why it is trying to connect to HTTP. Can you cat /opt/so/saltstack/default/salt/kibana/etc/kibana.,yml and paste it here?

**Reply by xaqpuyrthnbg** [reply]
Sure:

Default Kibana configuration from kibana-docker.
{%- set ES = salt['pillar.get']('manager:mainip', '') -%}
server.name: kibana
server.host: "0"
server.basePath: /kibana
elasticsearch.hosts: [ "https://{{ ES }}:9200" ]
elasticsearch.ssl.verificationMode: none
#kibana.index: ".kibana"
#elasticsearch.username: elastic
#elasticsearch.password: changeme
#xpack.monitoring.ui.container.elasticsearch.enabled: true
elasticsearch.requestTimeout: 90000
logging.dest: /var/log/kibana/kibana.log
telemetry.enabled: false
security.showInsecureClusterWarning: false
xpack.security.authc.providers:
  anonymous.anonymous1:
    order: 0
    credentials: "elasticsearch_anonymous_user"

**Reply by TOoSmOotH** [reply]
Can you do sudo docker logs so-kibana

**Reply by xaqpuyrthnbg** [reply]
No output when I ran the command.

**Comment by thunderwood19** [comment]
Had the same error yesterday as well.  What fixed mine was deleting the custom log stash pipeline yml's under /opt/so/saltstack/local/salt/logstash/pipelines/so/.  Then I ran salt \* state.apply (Im on a distributed net) and then I rebooted the master for good measure (unknown if nescessary).  Now I have errors with shards failing, but kibana works!

**Reply by TOoSmOotH** [reply]
Everything now uses SSL so your custom pipelines need to be modified to use https vs http.

**Reply by thunderwood19** [reply]
Yep...Had a brainfart moment. Luckily "https" to http was the only modification I had.

**Comment by TOoSmOotH** [comment, marked as answer]
https://github.com/elastic/kibana/issues/51007

Looks like you need to delete the latest .kibana index and restart kibana.

**Reply by xaqpuyrthnbg** [reply]
I have tried this before and it worked; however today I don't get anything when greping for the indices:

`curl -s localhost:9200/_cat/indices | grep kibana`

**Reply by TOoSmOotH** [reply]
You need to query elasticsearch not Kibana

**Reply by xaqpuyrthnbg** [reply]
Just confirming that you are referring to the fact that I had the incorrect port number?  I copy / pasted the wrong command into the thread here.  I do not get any results from the corrected command.

**Reply by TOoSmOotH** [reply]
curl -sk https://localhost:9200/_cat/indices

**Reply by xaqpuyrthnbg** [reply]
You did say "Everything now uses SSL..."  

It's all good now.

TY
